src/compositemark/errorbar.ts

```diff
@@ -1,42 +1,395 @@
+export function filterUnsupportedChannels(spec: GenericUnitSpec<Encoding<string>, ErrorBar | ErrorBarDef>): GenericUnitSpec<Encoding<string>, ErrorBar | ErrorBarDef> {
+  return {
+    ...spec,
+    encoding: reduce(spec.encoding, (newEncoding, fieldDef, channel) => {
+      if (supportedChannels.indexOf(channel) > -1) {
+        newEncoding[channel] = fieldDef;
+      } else {
+        log.warn(log.message.incompatibleChannel(channel, ERRORBAR));
+      }
+      return newEncoding;
+    }, {}),
+  };
+}
+
+export function normalizeErrorBar(spec: GenericUnitSpec<Encoding<string>, ErrorBar | ErrorBarDef>, config: Config): NormalizedLayerSpec {
+  spec = filterUnsupportedChannels(spec);
+
   // TODO: use selection
-  const {mark: _m, selection: _sel, projection: _p, encoding, ...outerSpec} = spec;
-  const {size: _s, ...encodingWithoutSize} = encoding;
-  const {x2: _x2, y2: _y2, ...encodingWithoutX2Y2} = encoding;
-  const {x: _x, y: _y, ...encodingWithoutX_X2_Y_Y2} = encodingWithoutX2Y2;
+  const {mark, encoding, selection, projection: _p, ...outerSpec} = spec;
+  const markDef: ErrorBarDef = isMarkDef(mark) ? mark : {type: mark};
+
+  // TODO(https://github.com/vega/vega-lite/issues/3702): add selection support
+  if (selection) {
+    log.warn(log.message.selectionNotSupported('boxplot'));
+  }
+
+  const center: ErrorBarCenter = markDef.center || config.errorbar.center;
+  const extent: ErrorBarExtent = markDef.extent || ((center === 'mean') ? config.errorbar.meanExtent : config.errorbar.medianExtent);
+
+  // add warning check in the test in this format.
+  // it('warn if size is data driven and autosize is fit', log.wrap((localLogger) => {
+  //   const spec = compile({
+  //     "data": {"values": [{"a": "A","b": 28}]},
+  //     "mark": "bar",
+  //     "autosize": "fit",
+  //     "encoding": {
+  //       "x": {"field": "a", "type": "ordinal"},
+  //       "y": {"field": "b", "type": "quantitative"}
+  //     }
+  //   }).spec;
+  //   assert.equal(localLogger.warns[0], log.message.CANNOT_FIX_RANGE_STEP_WITH_FIT);
+  //   assert.equal(spec.width, 200);
+  //   assert.equal(spec.height, 200);
+  // }));
+  if ((center === 'median') !== (extent === 'iqr')) {
+    log.warn(`${center} is not usually used with ${extent} for error bar.`);
+  }
 
-  if (!encoding.x2 && !encoding.y2) {
-    throw new Error('Neither x2 or y2 provided');
+  const orient: Orient = errorBarOrient(spec);
+  const {transform, continuousAxisChannelDef, continuousAxis, groupby, encodingWithoutContinuousAxis} = errorBarParams(spec, orient, center, extent);
+
+  const {size: _s, ...encodingWithoutSizeAndContinuousAxis} = encodingWithoutContinuousAxis;
+
+  const {scale, axis} = continuousAxisChannelDef;
+
+  function errorBarSinglePointPartMarksSpec(partName: ErrorBarPart, startingPosition: string = center, endingPosition:string = undefined) {
+    const markProperties: boolean | object = config.errorbar[partName];
```

Comment:
`partLayerMixins` already applies config (and also apply it before `opacity`). 

Remove these two constants and set

```
mark: getMarkNameOfErrorBarPart(partName)
```

below should solve your opacity problem.